### PR TITLE
V2.0 picturefill

### PR DESCRIPTION
**Pull request by jansepar**
TODO:
- ~~Needs tests~~
- ~~More fleshed out examples~~
- ~~Updated documentation~~
- Checkin alpha4

### CODE REVIEW

File: examples/capturing-picturepolyfill/index.html [hunk 1]
```diff
@@ -99,12 +99,11 @@
 <body class="foo">
     <p>Credit to scottjehl for his <a href="http://scottjehl.github.com/picturefill/">picturefill example</a></p>
     <p>This example has been forked and replicated using capturing to avoid the need for a &lt;noscript&gt; tag (view source to see for yourself).</p>
-    <p>Open up the network tab of your web inspector to see that the img element does not have it's asset downloaded.<p>
-    <picture>
-        <source src="/mobifyjs/examples/assets/images/small.jpg">
-        <source src="/mobifyjs/examples/assets/images/medium.jpg" media="(min-width: 450px)">
-        <source src="/mobifyjs/examples/assets/images/large.jpg" media="(min-width: 800px)">
-        <source src="/mobifyjs/examples/assets/images/extralarge.jpg" media="(min-width: 1000px)">
+    <p>Open up the network tab of your web inspector to see that the img element does not have it's asset downloaded.</p>
```
Comment (noahadams) [inline examples/capturing-picturepolyfill/index.html]: its, it's is a contraction for "it is"

**Comment by jansepar** [comment]
@noahadams alright, ramped up my tests and got things working across all of our supported browsers! This is ready for review.

File: src/external/picturefill.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        }
+        oldRenderCapturedDoc.apply(this, arguments);
+    }
+
+    return;
+}
+
+window.matchMedia = Utils.matchMedia(document);
+
+/*! Picturefill - Author: Scott Jehl, 2012 | License: MIT/GPLv2 */
```
Comment (noahadams) [inline src/external/picturefill.js]: a link to the GH page might be nice

File: src/external/picturefill.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    // Override renderCapturedDoc to disable img elements in picture elements
+    var oldRenderCapturedDoc = Capture.prototype.renderCapturedDoc;
+    Capture.prototype.renderCapturedDoc = function(options) {
+        // Change attribute of any img element inside a picture element
+        // so it does not load post-flood
+        var imgsInPicture = this.capturedDoc.querySelectorAll('picture img');
+        for (var i = 0, len = imgsInPicture.length; i < len; i++) {
+            var disableImg = imgsInPicture[i];
+            var srcAttr = window.Mobify && window.Mobify.prefix + 'src';
+            disableImg.setAttribute('data-orig-src', disableImg.getAttribute(srcAttr));
```
Comment (noahadams) [inline src/external/picturefill.js:15]: Why not just remove the element?

Comment (jansepar) [inline src/external/picturefill.js:15]: The picture polyfill uses the `img` element to display the image that is chosen using the branching logic. If we removed the `img` element, the picture polyfill would just create one again.

File: src/external/picturefill.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            disableImg.removeAttribute(srcAttr);
+        }
+        oldRenderCapturedDoc.apply(this, arguments);
+    }
+
+    return;
+}
+
+window.matchMedia = Utils.matchMedia(document);
```
Comment (noahadams) [inline src/external/picturefill.js]: this path never falls back to a native implementation if it exists (or is hooked up to the document we want)

**Comment by noahadams** [comment]
You wanna merge master into this given our changes from yesterday?

File: README.md [hunk 5]
```diff
@@ -25,11 +32,11 @@ Now, to build the library, simply run grunt:
 
 Open the ./build folder to see the generated Mobify.js libraries.
 
-Mobify.js uses Require.js mainly for dependency management, we don't use it for dynamic library loading when developing.
+Mobify.js uses Require.js mainly for dependency management, and doesn't use it for dynamic library loading when developing.
```
Comment (noahadams) [inline README.md]: Would be more correct to say "is built with AMD modules using almond"

Comment (jansepar) [inline README.md]: Yeah, I like that.

File: www/v2/docs/image-resizer.md [hunk 6]
```diff
@@ -5,63 +5,88 @@ title: Mobify.js Documentation
...
+- Determines support for WEBP and uses that on images whenever possible.
+Otherwise it defaults to the original image format.
 
-Our image resizing service backend service is free to use up to a certain number
-of views per month. If you plan on using this service on a website with high 
-amounts of traffic, feel free to visit our 
-[pricing page](http://www.mobify.com/pricing/) for more detail.
+**Note: The image resizer backend must have access to the images in order to resize
+them. If your development server is not publicly accessible, 
+ir.mobify.com will serve a 302 redirect back to the original image location.**
```
Comment (noahadams) [inline www/v2/docs/image-resizer.md]: ir0

File: www/v2/docs/image-resizer.md [hunk 7]
```diff
@@ -5,63 +5,88 @@ title: Mobify.js Documentation
...
+
+<div class="alert alert-block">
+    <p>Our image resizing service backend service is free to use up to a certain number
+    of views per month. If you plan on using this service on a website with high 
+    amounts of traffic, feel free to visit our
```
Comment (noahadams) [inline www/v2/docs/image-resizer.md]: "please"

File: www/v2/docs/image-resizer.md [hunk 8]
```diff
@@ -96,6 +121,69 @@ The same as `ResizeImages.resize` options.
         return "http://src.sencha.io/" + options.maxWidth + "/" + url  
     };
 
+## WEBP
+
+Using the Image Resizer API with Mobify.js, you can automatically have WEBP
```
Comment (noahadams) [inline www/v2/docs/image-resizer.md]: your images can be automatically converted to webp on browsers that support it.

**Comment by jansepar** [comment]
:+1: to your edits :)

**Comment by noahadams** [comment]
:+1:

**Pull request merged**